netty-reactive-streams/src/main/java/com/typesafe/netty/HandlerPublisher.java

```diff
@@ -203,32 +203,40 @@ private void provideSubscriber(Subscriber<? super T> subscriber) {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        verifyRegisteredWithRightExecutor(ctx);
+        // If the channel is not yet registered, then it's not safe to invoke any methods on it, eg read() or close()
+        // So don't provide the context until it is registered.
+        if (ctx.channel().isRegistered()) {
+            provideChannelContext(ctx);
+        }
```

Comment:
Because the channel will often have already been registered to the event loop.  In fact, there is not a 1:n relationship between channels and HandlerPublishers, each time you handle a request body in Play for example, you will add a HandlerPublisher to the pipeline to handle it.  The event loop only gets registered once, right when the connection is received, well before you even know that you're going to have a subscriber to handle the request body because there's no request yet.